src/otx/engine/engine.py

```diff
@@ -0,0 +1,407 @@
+        >>> engine = Engine(
+            model = OTXModel(...),
+            datamodule = OTXDataModule(...),
+        )
+    """
+
+    def __init__(
+        self,
+        *,
+        data_root: str | Path | None = None,
+        task: OTXTaskType | None = None,
+        work_dir: str | Path = "./otx-workspace",
+        datamodule: OTXDataModule | None = None,
+        model: OTXModel | str | None = None,
+        optimizer: OptimizerCallable | None = None,
+        scheduler: LRSchedulerCallable | None = None,
+        checkpoint: str | None = None,
+        device: DeviceType = DeviceType.auto,
+        **kwargs,
+    ):
+        """Initializes the Engine object.
+
+        Args:
+            data_root (str | Path): The root directory of the data.
+            task (OTXTaskType | None, optional): The task type. Defaults to None.
+            work_dir (str | Path, optional): The working directory. Defaults to "./otx-workspace".
+            datamodule (OTXDataModule | None, optional): The data module. Defaults to None.
+            model (OTXModel | str | None, optional): The model. Defaults to None.
+            optimizer (OptimizerCallable | None, optional): The optimizer. Defaults to None.
+            scheduler (LRSchedulerCallable | None, optional): The learning rate scheduler. Defaults to None.
+            checkpoint (str | None, optional): The checkpoint. Defaults to None.
+            device (DeviceType, optional): The device type. Defaults to DeviceType.auto.
+            **kwargs: Additional keyword arguments for pl.Trainer.
+        """
+        self.work_dir = work_dir
+        self.checkpoint = checkpoint
+        self.device = DeviceConfig(accelerator=device)
+        self._cache = TrainerArgumentsCache(
+            default_root_dir=self.work_dir,
+            accelerator=self.device.accelerator,
+            devices=self.device.devices,
+            **kwargs,
+        )
+
+        # [TODO] harimkang: It will be updated in next PR.
+        if not isinstance(model, OTXModel) or datamodule is None or optimizer is None or scheduler is None:
+            msg = "Auto-Configuration is not implemented yet."
+            raise NotImplementedError(msg)
+        self.datamodule: OTXDataModule = datamodule
+        self.task = self.datamodule.task
+
+        self._trainer: Trainer | None = None
+        self._model: OTXModel = model
+        self.optimizer: OptimizerCallable = optimizer
+        self.scheduler: LRSchedulerCallable = scheduler
+
+    """
+    General OTX Entry Points
+    """
+
+    def train(
+        self,
+        max_epochs: int = 10,
+        seed: int | None = None,
+        deterministic: bool = False,
+        precision: _PRECISION_INPUT | None = "32",
+        val_check_interval: int | float | None = 1,
+        callbacks: list[Callback] | Callback | None = None,
+        logger: Logger | Iterable[Logger] | bool | None = None,
+        resume: bool = False,
+        **kwargs,
```

Comment:
Engine.test, predict have already configured their arguments to receive a new dataloader. I think the point we should consider here is model. I thought the current OTX, apart from lightning's trainer, is that the engine provides training to exporting (+@) for one model. In this case, wouldn't it be more confusing for the user if we made them receive model for each entry point? 
Of course, users who are familiar with lightning may be more familiar with receiving it from functions such as fit and test, but I think it may be a bit confusing to put model in when it is created and also receive model in functions such as train for entry OTX user.

We'll have to think about it more, but for now I'm thinking 1 Engine = 1 model.
Get model args from train() may allow us to receive the model flexibly, as you intend, but we have to think about a more complicated overriding code flow than we might think (if the user keeps changing models, which model should we serve in some cases and which in others?)

What do you think?